### PR TITLE
Adapt UI to the new Host Inventory

### PR DESCRIPTION
**Pull request by mareklibra**
Former hardwareInfo (Introspection) is not used anymore.

### CODE REVIEW

**Comment by mareklibra** [comment]
---

**Comment by mareklibra** [comment]
@ori-amizur , can we get more data even from a virtual machine? 
- serial number can be libvirt's identifier or similar
- default value for speed should be 0

Is the `0.0.0.0` BMC address a hardcoded default? Can it be empty in that case?

**Comment by ori-amizur** [comment]
> @ori-amizur , can we get more data even from a virtual machine?
> 
> * serial number can be libvirt's identifier or similar
> * default value for speed should be 0
> 
> Is the `0.0.0.0` BMC address a hardcoded default? Can it be empty in that case?

@mareklibra We can provide the system uuid. (dmidecode -s system-uuid). In case of a virtual machine it is the same as the host-id  Wouldn't it be misleading if we put it in the place of serial number?
The returned value of the nic speed is provided from /sys/class/net/<nic-name>/speed without interpretation.  We can change the value to 0 in case it is -1.
The 0.0.0.0 is hardcoded default.  It is the same as provided by ironic python agent.  It can be changed to empty string.

**Comment by mareklibra** [comment]
Rebased

File: src/components/clusterConfiguration/HostsTable.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ type OpenRows = {
 };
 
 const columns = [
-  { title: 'Serial Number', transforms: [sortable], cellFormatters: [expandable] },
+  { title: 'Host name', transforms: [sortable], cellFormatters: [expandable] },
```
Comment (jtomasek) [inline src/components/clusterConfiguration/HostsTable.tsx:44]: I think this should be called 'Hostname'

File: src/components/clusterConfiguration/HostsTable.tsx [hunk 2]
```diff
@@ -67,7 +68,7 @@ const hostToHostTableRow = (openRows: OpenRows) => (host: Host): IRow => {
       // visible row
       isOpen: !!openRows[id],
       cells: [
-        id,
+        inventory.hostname,
```
Comment (jtomasek) [inline src/components/clusterConfiguration/HostsTable.tsx:71]: The hostname is not set until the host gets discovered.

```suggestion
        inventory.hostname || EMPTY,
```

Comment (mareklibra) [reply src/components/clusterConfiguration/HostsTable.tsx]: done

File: src/components/clusterConfiguration/HostsTable.tsx (hunk 1 above)
Comment (mareklibra) [reply src/components/clusterConfiguration/HostsTable.tsx]: done

File: src/components/hosts/HostsTable.tsx [hunk 3]
```diff
@@ -51,16 +52,16 @@ const columns = [
...
+  const inventory = stringToJSON<Inventory>(inventoryString) || {};
+  const { cores, memory, disk } = getHostRowHardwareInfo(inventory);
 
   return [
     {
       // visible row
       isOpen: !!openRows[id],
       cells: [
-        id,
+        inventory.hostname || EMPTY,
```
Comment (jtomasek) [inline src/components/hosts/HostsTable.tsx:64]: Sorry for not noticing it earlier, I think the EMPTY needs to be similar to EMPTY on hw columns?
```suggestion
        inventory.hostname || { title: DASH, sortableValue: '' },
```

**Pull request merged**